platform-test/base/unification.oz

```diff
@@ -0,0 +1,106 @@
+functor
+import
+   System(show:Show)
+export
+   Return
+define
+   ShouldHaveFailed = 'should have failed'
+   proc{MustFail X P}
+      skip
+      try
+         {P}
+         raise ShouldHaveFailed end
+      catch S then case S of !ShouldHaveFailed then
+         raise '  '(X ShouldHaveFailed) end
+      [] _ then
+         skip
+      end end
```

Comment:
I think "end end" should be avoided.
Catch also does pattern matching iirc, could you use that?
Also a unification failure is a very specific exception, so probably only a failure should be caught.